src/vm/functions/database.rs

```diff
@@ -497,3 +497,52 @@ pub fn special_get_block_info(
 
     Ok(Value::some(result)?)
 }
+
+pub fn special_get_burn_block_info(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    // (get-block-info? property-name block-height-int)
+    runtime_cost(ClarityCostFunction::Unimplemented, env, 0)?;
+
+    check_argument_count(2, args)?;
+
+    // Handle the block property name input arg.
+    let property_name = args[0]
+        .match_atom()
+        .ok_or(CheckErrors::GetBlockInfoExpectPropertyName)?;
+
+    let block_info_prop = BurnBlockInfoProperty::lookup_by_name(property_name)
+        .ok_or(CheckErrors::GetBlockInfoExpectPropertyName)?;
+
+    // Handle the block-height input arg clause.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
+        Value::UInt(result) => Ok(result),
+        x => Err(CheckErrors::TypeValueError(TypeSignature::UIntType, x)),
+    }?;
+
+    let height_value = match u32::try_from(height_value) {
+        Ok(result) => result,
+        _ => return Ok(Value::none()),
```

Comment:
Do we want to be returning an error here? Returning Value::none() is not too informative of what went wrong?

Reply:
This is copy-pasted.. I guess the assumption was that we would never overflow u32 in practice. I.e., if `result` is u128, and we can't cast to u32, then it is bigger than u32::MAX.

Can we assume that?

Reply:
Right, but this branch represents the error case of overflow, so I feel like the result should be a corresponding error.

Reply:
This would have to be a user facing error? which one would it be? https://github.com/blockstack/stacks-blockchain/blob/4650aae6ffb111d487ce324fd8591c529e60c24e/src/vm/analysis/errors.rs#L27

Reply:
I think `CheckErrors::ValueTooLarge` makes sense here

Reply:
You're right, that error type doesn't make sense. 

I'm alright with returning `Value::none()` here.